components/http/okHttp/src/main/java/com/microsoft/kiota/http/middleware/options/UserAgentHandlerOption.java

```diff
@@ -0,0 +1,69 @@
+package com.microsoft.kiota.http.middleware.options;
+
+import java.util.Objects;
+
+import javax.annotation.Nonnull;
+
+import com.microsoft.kiota.RequestOption;
+
+/** The options for the user agent handler */
+public class UserAgentHandlerOption implements RequestOption {
+	private boolean enabled = true;
+	@Nonnull
+	private String productName = "kiota-java";
+	@Nonnull
+	private String productVersion = "0.0.10";
```

Comment:
Is it possible to read this at runtime? Otherwise we may have to keep updating it as well

Reply:
Not as far as my research yielded. The version number is merely metadata on the repository (package feed) but not part of the jar itself like it is in dotnet.

Reply:
Actually this might be worth trying https://stackoverflow.com/a/21725706

Reply:
after trying it, it will only work when packaged as a jar and added to the project from the feed (with its manifest). That's already a problem in itself for the unit test. But there's also a packaging technique called fat jars in Java where people mush all the dependencies and their code in a single jar, loosing all the manifests in the process.
This means the package version would be outside of our control.

Reply:
Thanks for clearing this one up. Resolving for now.